auth/auth_utilities.py

```diff
@@ -33,4 +36,203 @@ def from_list(scopes_list: list[str]) -> str:
+        Initialize with a string key for hashing.
+        
+        :param key: String to use as a salt/key in the hash generation
+        """
+        self._key = key
+    
+    def generate_hash(self, data_dict: dict) -> str:
+        """
+        Generate a one-way hash digest using the key as a salt and provided dictionary.
+        
+        :param data_dict: Dictionary to hash
+        :return: URL-safe base64 encoded hash digest
+        """
+        # Convert dictionary to a sorted JSON string for consistent hashing
+        serialized_data = json.dumps(data_dict, sort_keys=True)
+        
+        # Create a hash object and update it with the key first (as a salt)
+        hash_obj = hashlib.sha256()
+        hash_obj.update(self._key.encode())
+        
+        # Then update with the serialized data
+        hash_obj.update(serialized_data.encode())
+        
+        # Get the digest
+        hash_digest = hash_obj.digest()
+        
+        # Return URL-safe base64 encoded hash (with padding removed)
+        return base64.urlsafe_b64encode(hash_digest).rstrip(b'=').decode()
+    
+    def verify_hash(self, data_dict: dict, hash_to_verify: str) -> bool:
+        """
+        Verify if the provided dictionary matches the given hash.
+        
+        :param data_dict: Dictionary to check against the hash
+        :param hash_to_verify: Hash string to verify against
+        :return: True if the dictionary generates the same hash, False otherwise
+        """
+        # Generate hash from the provided dictionary        
+        computed_hash = self.generate_hash(data_dict)        
+        
+        # Compare the computed hash with the provided hash
+        return computed_hash == hash_to_verify
+    
+    def __repr__(self):
+        return f"HashGenerator(key={self._key!r})"
+    
+
+class ConsentCookieGenerator:
+    """Class to generate a consent cookie."""
+    
+    def __init__(self, application_id:str, 
+                 redirect_uri: str, 
+                 scopes: str, 
+                 authorization_code: str, 
+                 state: str, 
+                 cookie_max_age: int, 
+                 auth_key: str = None):
+        """
+        Initialize with parameters to generate a consent cookie.
+        
+        :param application_id: ID of the application
+        :param redirect_uri: URI to redirect after consent
+        :param scopes: Scopes requested by the application
+        :param authorization_code: Authorization code for OAuth flow
+        :param state: State parameter for OAuth flow
+        :param cookie_max_age: Max age of the cookie in seconds
+        """
+        self.application_id = application_id
+        self.redirect_uri = redirect_uri
+        self.scopes = scopes
+        self.authorization_code = authorization_code
+        self.state = state
+        self.cookie_max_age = cookie_max_age
+        self.hashing_util = HashSignatureUtility(auth_key)
+
+    def generate_cookie(self) -> str:
```

Comment:
Since this class only has one method, should we consider just making it a static function?